### DISCUSSION TITLE
MariaDB Stored Procedure

### DISCUSSION THREAD

**Discussion by ricklangiii** [Budibase/budibase#14055] [Help] [answered]
New user here at my wits end.  I've searched the forum and stumped ChatGPT and Claude.  I have a Stored Procedure in MariaDB named GetOppDetails which expects to receive an opp_id.  I've setup a query in Budibase to CALL GetOppDetails({{ opp_id }});m with a binding on opp_id.  This query executes properly in the MariaDB console (HeidiSQL).  When I run the query in Budibase, I receive the results AND metadata.  This metadata seems to be tripping up Budibase.  Any ideas?  Below is the output (Query Results) from Budibase.  I should add, I've tried the default transformer of 'return data' as well as 'return data.map(element => ({
...element,
	filter: params.opp_id
}))'
and both yield the same output.
Output:
[
  {
    "0": {
      "item_name": "cost",
      "item_description": "desc",
      "quantity": 5,
      "unit_price_per_mo": "1.00",
      "month_1": "5.00",
      "month_2": "5.00",
      "month_3": "5.00",
      "month_4": "5.00",
      "month_5": "5.00",
      "month_6": "5.00",
      "month_7": "5.00",
      "month_8": "5.00",
      "month_9": "5.00",
      "month_10": "5.00",
      "month_11": "5.00",
      "month_12": "5.00",
      "month_13": "5.00",
      "month_14": "5.00",
      "month_15": "5.00",
      "month_16": "5.00",
      "month_17": "5.00",
      "month_18": "5.00",
      "month_19": "5.00",
      "month_20": "5.00",
      "month_21": "5.00",
      "month_22": "5.00",
      "month_23": "5.00",
      "month_24": "5.00",
      "month_25": "5.00",
      "month_26": "5.00",
      "month_27": "5.00",
      "month_28": "5.00",
      "month_29": "5.00",
      "month_30": "5.00",
      "month_31": "5.00",
      "month_32": "5.00",
      "month_33": "5.00",
      "month_34": "5.00",
      "month_35": "5.00",
      "month_36": "5.00",
      "total_cost": "180.00"
    },
    "1": {
      "item_name": "cost",
      "item_description": "desc",
      "quantity": 1,
      "unit_price_per_mo": "5.00",
      "month_1": null,
      "month_2": null,
      "month_3": null,
      "month_4": null,
      "month_5": null,
      "month_6": "5.00",
      "month_7": null,
      "month_8": null,
      "month_9": null,
      "month_10": null,
      "month_11": null,
      "month_12": null,
      "month_13": null,
      "month_14": null,
      "month_15": null,
      "month_16": null,
      "month_17": null,
      "month_18": null,
      "month_19": null,
      "month_20": null,
      "month_21": null,
      "month_22": null,
      "month_23": null,
      "month_24": null,
      "month_25": null,
      "month_26": null,
      "month_27": null,
      "month_28": null,
      "month_29": null,
      "month_30": null,
      "month_31": null,
      "month_32": null,
      "month_33": null,
      "month_34": null,
      "month_35": null,
      "month_36": null,
      "total_cost": "5.00"
    },
    "2": {
      "item_name": "Total",
      "item_description": null,
      "quantity": null,
      "unit_price_per_mo": null,
      "month_1": "5.00",
      "month_2": "5.00",
      "month_3": "5.00",
      "month_4": "5.00",
      "month_5": "5.00",
      "month_6": "10.00",
      "month_7": "5.00",
      "month_8": "5.00",
      "month_9": "5.00",
      "month_10": "5.00",
      "month_11": "5.00",
      "month_12": "5.00",
      "month_13": "5.00",
      "month_14": "5.00",
      "month_15": "5.00",
      "month_16": "5.00",
      "month_17": "5.00",
      "month_18": "5.00",
      "month_19": "5.00",
      "month_20": "5.00",
      "month_21": "5.00",
      "month_22": "5.00",
      "month_23": "5.00",
      "month_24": "5.00",
      "month_25": "5.00",
      "month_26": "5.00",
      "month_27": "5.00",
      "month_28": "5.00",
      "month_29": "5.00",
      "month_30": "5.00",
      "month_31": "5.00",
      "month_32": "5.00",
      "month_33": "5.00",
      "month_34": "5.00",
      "month_35": "5.00",
      "month_36": "5.00",
      "total_cost": "185.00"
    },
    "filter": "1"
  },
  {
    "fieldCount": 0,
    "affectedRows": 2,
    "insertId": 0,
    "info": "",
    "serverStatus": 34,
    "warningStatus": 0,
    "changedRows": 0,
    "filter": "1"
  }
]

I'm sure it's something simple - I'm a newb in this environment!  Any help is appreciated.

**Comment by ConorWebb96** [comment]
Hey @ricklangiii,

Whats the goal with the above data? Are you wanting this to be displayed in a table?

**Reply by ConorWebb96** [reply, marked as answer]
If you are just looking to output the array objects within objects, that hold your monthly expenses over 36months heres some JS that you'll need to do it. 

```js
const array = data;

const transformArray = (arr) => {
    const firstObject = arr[0];
    const newArray = [];
    for (const key in firstObject) {
        if (firstObject.hasOwnProperty(key) && !isNaN(key)) {
            newArray.push(firstObject[key]);
        }
    }

    return newArray;
};
const result = transformArray(array);

return result;
```

Put the above in your transformer and it should output everything correctly. I've used an api to quickly setup a response with your data structure can see this response below.

**Data view**
<img width="1250" alt="Screenshot 2024-07-01 at 09 17 36" src="https://github.com/Budibase/budibase/assets/126772285/43438222-b9c5-4d1b-9b5e-405c16cf0626">

**Design view.**
<img width="1417" alt="Screenshot 2024-07-01 at 09 18 10" src="https://github.com/Budibase/budibase/assets/126772285/1dd96cbc-2e66-465e-b58e-b861f8b3ca22">

I hope this helps.

**Reply by ricklangiii** [reply]
thank you!
